test/src/org/omegat/filters/TestFilterBase.java

```diff
@@ -299,22 +299,29 @@ protected void translate(AbstractFilter filter, String filename) throws Exceptio
     }
 
     /**
-     * Helper function for testing the translateFile method of a filter. Translation equals the source.
+     * Helper function for testing the translateFile method of a filter.
+     * Translation equals the source when monolingual filter or monolingual option.
      * Translation is written to {@link #outFile}.
      * @param filter the filter to test
      * @param filename the file to use as input for the filter
      * @param config the filter options/config to use
      * @throws Exception when the filter throws an exception
      */
     protected void translate(AbstractFilter filter, String filename, Map<String, String> config) throws Exception {
+        boolean allowBlank;
+        if (filter.isBilingual()) {
+            allowBlank = !"true".equalsIgnoreCase(config.get("monolingualFormat"));
```

Comment:
The POFilter specific logic should be in POFilterTest class as overriding base class method.